kafka-client/src/main/java/dev/responsive/kafka/store/ResponsivePartitionedStore.java

```diff
@@ -130,22 +142,100 @@ public void init(final StateStoreContext context, final StateStore root) {
           new BytesKeySpec(),
           config
       );
-      buffer.init();
 
-      open = true;
+      if (isActive()) {
+        registration = new ResponsiveStoreRegistration(
+            name.kafkaName(),
+            topicPartition,
+            buffer::flush,
+            this::storedOffset
+        );
+      } else {
+        registration = new ResponsiveStoreRegistration(
+            name.kafkaName(),
+            topicPartition,
+            buffer::flush,
+            this::storedOffset,
+            this::maybeTransitionToActive
```

Comment:
[major] am I reading this correctly in that if we started off with `isActive() == true` that we'll never initialize the buffer? (it seems like it only happens in the `maybeTransitionToActive` callback